File: TyreVibes/Screen/OnBoarding/OnBoardingScreen.swift
import SwiftUI

struct OnboardingView: View {
    // Stato per tracciare la pagina corrente dell'onboarding
    @State private var currentPage = 0
    
    // Array di contenuti dell'onboarding
    let onboardingData = [
        OnboardingPage(
            title: "Lorem ipsum sem arcu placerat",
            description: "Ciao, sono Matteo. Vivo a Milano. Vivo a Milano. Vivo a Milano.",
            image: "imageOnBoardingScreen1"
        ),
        OnboardingPage(
            title: "Etiam vitae consectetur",
            description: "Lavoro come sviluppatore di app. Mi piace creare interfacce intuitive.",
            image: "onboarding2"
        ),
        OnboardingPage(
            title: "Nullam fermentum",
            description: "Amo viaggiare ed esplorare nuove città. La fotografia è la mia passione.",
            image: "onboarding3"
        )
    ]
    
    // Stato per gestire l'uscita dall'onboarding
    @State private var isOnboardingComplete = false
    
    var body: some View {
        ZStack {
            // Sfondo nero
            Color.black.edgesIgnoringSafeArea(.all)
            
            // Se l'onboarding è completo, mostra la vista principale
           if isOnboardingComplete {
                WelcomeScreen()
                    .transition(.asymmetric(insertion: .move(edge: .trailing).combined(with: .opacity),
                                            removal: .move(edge: .leading).combined(with: .opacity)))
                    .animation(.easeInOut(duration: 0.6), value: isOnboardingComplete)
            } else {
                VStack {
                    Spacer()
                    
                    // Area semicircolare nella parte superiore (può essere un'immagine o una formwa)
                    Circle()
                        .fill(Color.customGray)
                        .frame(width: UIScreen.main.bounds.width * 1.5, height: UIScreen.main.bounds.width * 1.5)
                        .position(x: UIScreen.main.bounds.width / 2, y: -UIScreen.main.bounds.width * -0.3)
                        .edgesIgnoringSafeArea(.top)
                    
                    // Immagine centrata sopra il cerchio
                    Image(onboardingData[currentPage].image)
                        .resizable()
                        .scaledToFit()
                        .frame(height: 200)
                        .padding(.top, -UIScreen.main.bounds.width * 0.7)
                    
                    Spacer()
                    
                    Text(onboardingData[currentPage].title)
                        .font(.customFont(size: 40, weight: .semibold))
                        .overlay(
                            LinearGradient(
                                gradient: Gradient(colors: [Color(hex: "#F78F18"), Color(hex: "#FF4C81")]),
                                startPoint: .leading,
                                endPoint: .trailing
                            )
                        )
                        .mask(
                            Text(onboardingData[currentPage].title)
                                .font(.system(size: 40, weight: .bold))
                        )
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 16)
                    
                    // Descrizione in testo bianco
                    Text(onboardingData[currentPage].description)
                        .font(.customFont(size: 16, weight: .regular))
                        .foregroundColor(.white)
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 24)
                        .padding(.top, 8)
                    
                    Spacer()
                    
                    // Indicatori di pagina
                    HStack(spacing: 4) {
                        ForEach(0..<onboardingData.count, id: \.self) { index in
                            Capsule()
                                .fill(currentPage == index ? Color.white : Color(hex: "808080"))
                                .cornerRadius(10)
                                .frame(width: currentPage == index ? 52 : 12, height: currentPage == index ? 6 : 12,)
                                .animation(.spring(), value: currentPage)
                        }
                    }
                    .padding(.bottom, 50)
                    
                    // Pulsanti Skip e Continue
                    HStack {
                        // Pulsante Skip
                        Button(action: {
                            withAnimation {
                                isOnboardingComplete = true
                            }
                        }) {
                            Text("Skip")
                                .font(.customFont(size: 18, weight: .regular))
                                .foregroundColor(.white)
                                .padding()
                        }
                        
                        Spacer()
                        
                        // Pulsante Continue
                        Button(action: {
                            withAnimation {
                                if currentPage < onboardingData.count - 1 {
                                    currentPage += 1
                                } else {
                                    isOnboardingComplete = true
                                }
                            }
                        })
                        {
                            Text("Continue")
                                .font(.customFont(size: 16, weight: .semibold))
                                .fontWeight(.semibold)
                                .foregroundColor(.white)
                                .frame(width: 134, height: 50)
                                .background(Color.customBitterSweet)
                                .cornerRadius(25)
                        }
                        
                    }
                    .padding(.horizontal, 24)
                    .padding(.bottom, 2)
                }
            }
        }
    }
}

// Struttura per i dati di ogni pagina dell'onboarding
struct OnboardingPage {
    let title: String
    let description: String
    let image: String
}

// Definizione di ContentView come placeholder
struct ContentView: View {
    var body: some View {
        Text("TyreVibes")
    }
}

// Anteprima della vista
struct OnboardingView_Previews: PreviewProvider {
    static var previews: some View {
        OnboardingView()
    }
}

File: TyreVibes-Android/app/src/main/java/com/tyrevibes/android/features/onboarding/OnboardingScreen.kt
package com.tyrevibes.android.features.onboarding

import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tyrevibes.android.R
import com.tyrevibes.android.ui.theme.CustomBackgroundColor
import com.tyrevibes.android.ui.theme.CustomBitterSweet
import com.tyrevibes.android.ui.theme.TyreVibesTheme
import com.tyrevibes.android.ui.theme.soraFontFamily
import kotlinx.coroutines.launch

data class OnboardingPage(
    val title: String,
    val description: String,
    val imageRes: Int
)

val onboardingPages = listOf(
    OnboardingPage(
        title = "Lorem ipsum sem arcu placerat",
        description = "Ciao, sono Matteo. Vivo a Milano. Vivo a Milano. Vivo a Milano.",
        imageRes = R.drawable.onboarding_1
    ),
    OnboardingPage(
        title = "Etiam vitae consectetur",
        description = "Lavoro come sviluppatore di app. Mi piace creare interfacce intuitive.",
        imageRes = R.drawable.ic_google // Placeholder
    ),
    OnboardingPage(
        title = "Nullam fermentum",
        description = "Amo viaggiare ed esplorare nuove città. La fotografia è la mia passione.",
        imageRes = R.drawable.ic_apple // Placeholder
    )
)

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun OnboardingScreen(
    onOnboardingComplete: () -> Unit
) {
    val pagerState = rememberPagerState(pageCount = { onboardingPages.size })
    val scope = rememberCoroutineScope()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(CustomBackgroundColor),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        HorizontalPager(
            state = pagerState,
            modifier = Modifier.weight(1f)
        ) { page ->
            OnboardingPageContent(page = onboardingPages[page])
        }

        Row(
            Modifier
                .height(50.dp)
                .fillMaxWidth(),
            horizontalArrangement = Arrangement.Center
        ) {
            repeat(onboardingPages.size) { iteration ->
                val color = if (pagerState.currentPage == iteration) Color.White else Color.Gray
                Box(
                    modifier = Modifier
                        .padding(2.dp)
                        .clip(CircleShape)
                        .background(color)
                        .size(12.dp)
                )
            }
        }

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp, vertical = 16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            TextButton(onClick = onOnboardingComplete) {
                Text(
                    text = "Skip",
                    color = Color.White,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Normal
                )
            }

            Button(
                onClick = {
                    if (pagerState.currentPage < onboardingPages.size - 1) {
                        scope.launch {
                            pagerState.animateScrollToPage(pagerState.currentPage + 1)
                        }
                    } else {
                        onOnboardingComplete()
                    }
                },
                shape = RoundedCornerShape(25.dp),
                colors = ButtonDefaults.buttonColors(containerColor = CustomBitterSweet),
                modifier = Modifier.size(width = 134.dp, height = 50.dp)
            ) {
                Text(
                    text = "Continue",
                    color = Color.White,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold
                )
            }
        }
    }
}

@Composable
fun OnboardingPageContent(page: OnboardingPage) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp)
    ) {
        Image(
            painter = painterResource(id = page.imageRes),
            contentDescription = null,
            modifier = Modifier
                .size(200.dp)
                .padding(bottom = 40.dp)
        )
        Text(
            text = page.title,
            fontFamily = soraFontFamily,
            fontWeight = FontWeight.SemiBold,
            fontSize = 32.sp, // Reduced size for better fit
            color = Color.White,
            textAlign = TextAlign.Center
        )
        Spacer(modifier = Modifier.height(16.dp))
        Text(
            text = page.description,
            fontFamily = soraFontFamily,
            fontWeight = FontWeight.Normal,
            fontSize = 16.sp,
            color = Color.Gray,
            textAlign = TextAlign.Center
        )
    }
}

@Preview
@Composable
fun OnboardingScreenPreview() {
    TyreVibesTheme {
        OnboardingScreen(onOnboardingComplete = {})
    }
}
